core/history/history.go

```diff
@@ -174,6 +183,252 @@ func (h *history) Versions(id domain.FullID, lastVersionId string, limit int) (r
 	return
 }
 
+func (h *history) DiffVersions(req *pb.RpcHistoryDiffVersionsRequest) ([]*pb.EventMessage, *model.ObjectView, error) {
+	id := domain.FullID{
+		ObjectID: req.ObjectId,
+		SpaceID:  req.SpaceId,
+	}
+	previousState, _, _, err := h.buildState(id, req.PreviousVersion)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get state of versions %s: %w", req.PreviousVersion, err)
+	}
+
+	currState, sbType, _, err := h.buildState(id, req.CurrentVersion)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get state of versions %s: %w", req.CurrentVersion, err)
+	}
+
+	currState.SetParent(previousState)
+	msg, _, err := state.ApplyState(currState, false)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get history events for versions %s, %s: %w", req.CurrentVersion, req.PreviousVersion, err)
+	}
+
+	historyEvents := getHistoryEvents(msg)
+	spc, err := h.spaceService.Get(context.Background(), id.SpaceID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get space: %w", err)
+	}
+	dependentObjectIDs := objectlink.DependentObjectIDs(currState, spc, true, true, false, true, false)
+	metaD, err := h.objectStore.QueryByID(dependentObjectIDs)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get dependencies: %w", err)
+	}
+
+	details := make([]*model.ObjectViewDetailsSet, 0, len(metaD))
```

Comment:
you allocate an array and then increase it by one, consider changing lines, so appending would come first

Reply:
Fixed